os/board/common/partitions.c

```diff
@@ -163,6 +163,21 @@ static void move_to_next_part(const char **par)
 		while (*(*par)++ != ',');
 }
 
+int get_partition_num(char *part)
+{
+	int partno = 0;
+
+	char *partname = CONFIG_FLASH_PART_NAME;
+	while (*partname) {
+		if (!strncmp(partname, part, strlen(part) + 1)) {
```

Comment:
You should check it with `strlen(part)` and then you should check `,` at next step. Because user will not give the part with `,`.
I mean user will deliver "userfs" not "userfs,".
Then this will return always false.